### PR TITLE
DASH-820 Update public api add user registration

### PR DESCRIPTION
**Pull request by sardonyxwt**

### CODE REVIEW

File: server/src/services/publicApi/BuyAddress.mjs [hunk 1]
```diff
@@ -79,27 +104,28 @@ export default class BuyAddress extends Base {
...
-    const orders = await Order.findAll({
-      where: {
-        publicKey,
-        refProfileId: refProfile.id,
-      },
-      include: [OrderItem],
-    });
+    const domainFromChain = await fioApi.getFioDomain(fioDomain).then(formatChainDomain);
+
+    if (domainFromChain && domainFromChain.expiration * 1000 < Date.now()) {
```
Comment (trukhilio) [inline server/src/services/publicApi/BuyAddress.mjs:109]: Please, use `isDomainExpired` function here

File: server/src/services/publicApi/BuyAddress.mjs [hunk 2]
```diff
@@ -109,76 +135,68 @@ export default class BuyAddress extends Base {
...
-    if (domain) {
+    if (domainFromChain) {
       return generateErrorResponse(this.res, {
         error: 'Domain already registered',
         errorCode: PUB_API_ERROR_CODES.DOMAIN_IS_ALREADY_REGISTERED,
         statusCode: HTTP_CODES.BAD_REQUEST,
       });
     }
 
+    const [user] = await getExistUsersByPublicKeyOrCreateNew(publicKey, referralCode);
```
Comment (trukhilio) [inline server/src/services/publicApi/BuyAddress.mjs:195]: Why do you crate user here it's already made before?

**Review by trukhilio — APPROVED** [review]
Check comments

File: server/src/services/publicApi/BuyAddress.mjs (hunk 2 above)
Comment (sardonyxwt) [reply server/src/services/publicApi/BuyAddress.mjs:195]: Here I create user if type === 'domain' before I create user for type ===  'account'. I create user near last operation

Comment (trukhilio) [reply server/src/services/publicApi/BuyAddress.mjs:195]: I see. Thanks

**Review by trukhilio — APPROVED** [review]

**Pull request merged**